### PR TITLE
Move code into repo

### PR DESCRIPTION
**Pull request by dastels**

### CODE REVIEW

**Comment by dastels** [comment]
Um... why isn't travis running checks on this?

**Comment by ladyada** [comment]
there's no code you commited?

**Comment by dastels** [comment]
OK... I've had some git weirdness happening with the one.  I think it's good now (and it's checking).

**Comment by brentru** [comment]
@dastels  Ran into an error with a PyGamer running CircuitPython 4.1.0-beta and the simpletest example in this PR.

```
Auto-reload is on. Simply save files over USB to run them or enter REPL to disable.
code.py output:
Setting up SD card
Building sample bitmap and palette
Saving bitmap
Error saving bitmap
Traceback (most recent call last):
  File "code.py", line 50, in <module>
  File "/lib/adafruit_bitmapsaver.py", line 115, in save_bitmap
  File "/lib/adafruit_bitmapsaver.py", line 112, in save_bitmap
  File "/lib/adafruit_bitmapsaver.py", line 84, in _write_pixels
TypeError: 'Palette' object is not subscriptable
OSError
```

**Comment by dastels** [comment]
Yes. It requires 5.0

**Comment by brentru** [comment]
@dastels  Ok, you update the README.rst to reflect the requirement for CircuitPython 5.0?

File: adafruit_bitmapsaver.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                buffer_index += 1
+        output_file.write(row_buffer)
+
+def save_bitmap(bitmap, palette, file_or_filename):
+    """Save a bitmap (using an associated palette) to a 24 bit per pixel BMP file.
+
+    :param bitmap: the displayio.Bitmap to save
+    :param palette: the displayio.Palette to use for looking up colors in the bitmap
+    """
+    print("Saving bitmap")
```
Comment (brentru) [inline adafruit_bitmapsaver.py]: Library should not print.

File: adafruit_bitmapsaver.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    if not isinstance(palette, Palette):
+        raise ValueError('palette')
+    try:
+        if isinstance(file_or_filename, str):
+            output_file = open(file_or_filename, 'wb')
+        else:
+            output_file = file_or_filename
+
+        filesize = 54 + bitmap.height * _bytes_per_row(bitmap)
+        # print('Bitmap height: ', bitmap.height)
```
Comment (brentru) [inline adafruit_bitmapsaver.py]: remove or add a `debug` kwarg

File: adafruit_bitmapsaver.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        # print('Bitmap width: ', bitmap.width)
+        # print('Filesize: ', filesize)
+        _write_bmp_header(output_file, filesize)
+        _write_dib_header(output_file, bitmap)
+        _write_pixels(output_file, bitmap, palette)
+    except Exception:
+        print('Error saving bitmap')
```
Comment (brentru) [inline adafruit_bitmapsaver.py]: Could you `raise` instead of printing here?

File: adafruit_bitmapsaver.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        raise
+    else:
+        output_file.close()
+    print("Finished saving bitmap")
```
Comment (brentru) [inline adafruit_bitmapsaver.py]: `print` used again. Possibly return if bitmap was saved successfully.

File: adafruit_bitmapsaver.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+def save_bitmap(bitmap, palette, file_or_filename):
+    """Save a bitmap (using an associated palette) to a 24 bit per pixel BMP file.
+
+    :param bitmap: the displayio.Bitmap to save
+    :param palette: the displayio.Palette to use for looking up colors in the bitmap
+    """
+    print("Saving bitmap")
+    if not isinstance(bitmap, Bitmap):
+        raise ValueError('bitmap')
```
Comment (brentru) [inline adafruit_bitmapsaver.py]: Could you make these exceptions more verbose as to why the file is not accepted?

File: adafruit_bitmapsaver.py [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+import gc
+import struct
+import board
+from displayio import Bitmap, Palette, Display
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_BitmapSaver.git"
+
+#pylint:disable=line-too-long,broad-except,redefined-outer-name,invalid-name
```
Comment (ladyada) [inline adafruit_bitmapsaver.py]: hiya don't disable these, they're there for a reason! please lint the code so these errors don't come up.

**Pull request merged**